src/api/index.ts

```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from 'react';
+import axios, { AxiosResponse } from 'axios';
+
+enum RequestMethods {
+  Get = 'GET',
+  Post = 'POST',
+  Put = 'PUT',
+  Delete = 'DELETE'
+}
+
+interface RequestParams {
+  method: RequestMethods,
+  uri: string,
+  data?: any,
+  config?: any
+}
+
+async function request(method: RequestMethods, uri: string, data = {}, config = {}) {
+  const requestConfig = { ...config, method, uri };
+  const response = await axios(requestConfig);
+
+  return response;
+}
+
+const req = {
+  get: request.bind(null, RequestMethods.Get),
```

Comment:
`bind`는 함수를 Constructor로 사용할 때 `this`를 바꿔주기 위해서 사용하는 녀석이라 지금 상황에서는 맞지 않는 것 같아유...! 차라리 함수를 반환하는 다른 함수를 사용하는 것이 좋을 것 같습니다.

그리고 HTTP 스펙 상 GET과 DELETE에는 Request Body를 담을 수 없으니 참고해주세용!


```suggestion
  get: (uri: string, config = {}) => request(RequestMethods.Get, uri, config),
```